trackintel/preprocessing/triplegs.py

```diff
@@ -164,7 +162,7 @@ def generate_trips(spts, tpls, gap_threshold=15, add_geometry=True):
 
     trips_grouper = spts_tpls_no_act.groupby("temp_trip_id")
     trips = trips_grouper.agg(
-        {"user_id": "mean", "started_at": min, "finished_at": max, "type": list, "spts_tpls_id": list}
+        {"user_id": "first", "started_at": "first", "finished_at": "last", "type": list, "spts_tpls_id": list}
```

Comment:
I am not sure about "first" and "last" for timestamps. Isn't it safer to use "min" and "max" instead of assuming that the first timestamp is also the earliest?

Reply:
Both are safe to use.
In the groupby pandas documentation they write ([link](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html)):
> Groupby preserves the order of rows within each group.

And as we already sorted beforehand we can replace an O(N) with a O(1), but I think the advantage is marginal to be honest. ^^ I just liked it more but if you find it clearer the other way I would change it back.

Reply:
I can understand the "first" in started_at (sorted as you said), but am not sure about the "last" for finished_at. 

We just sorted according to the started_at columns and it could happen that the last finished_at is not the max finished_at. This means of course there is overlapping time and something is wrong with the input data, but I have encountered this in real-world tracking datasets. Perhaps this requires the user to filter these incorrect records first, but as the generate function we cannot assume that the user has done that. 

So I would say we remain "max" for the finished_at. Maybe I am thinking too much, what do you think?

Reply:
Yes, I agree. I am also on the better save than sorry side. Also if I think that the solution of @bifbof is probably the smarter way to do it I became quite anxious given all the bug reports ;-)

So we agree on the following?
- "user_id": "first"
- "started_at": min
- "finished_at": max